### PR TITLE
Graph notifications docs

### PR DESCRIPTION
**Pull request by leixu2046**
Adding Graph notifications concept and beta API docs.

Made some changes to existing Rome overview (both conceptual overview and the API overview under resource), as well as the permission model page and the change log page. 

The new files should be in this hierarchy inside the ToC but Laura please chime in if you have suggestions:

Conceptual doc:
concepts/graph-notifications-concept-overview.md should be a new node under https://developer.microsoft.com/en-us/graph/docs/concepts/cross-device-concept-overview, in parallel to the Activity Feed and Device Relay overviews;

API doc: 
api-reference/beta/resources/projectrome-graph-notifications-api-overview.md should be a new node under https://developer.microsoft.com/en-us/graph/docs/api-reference/beta/resources/project_rome_overview; 

api-reference/beta/resources/projectrome_notification.md should be under api-reference/beta/resources/projectrome-graph-notifications-api-overview.md

api-reference/beta/api/projectrome_post_notification.md should be under api-reference/beta/resources/projectrome_notification.md

In summary, just like how we list under Beta reference today:
Cross-device experience
- Activity Feed
--- Activity
----- Create or replace an activity

We should also have:
- Microsoft Graph Notifications
--- Notification
----- Create and publish a notification

Thanks,
Lei

### CODE REVIEW

File: api-reference/beta/resources/projectrome_notification.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+|:------|:------------|:-----------|
+|[Create notification](../api/projectrome_post_notification.md) | [notification](projectrome_notification.md) |Creates and publish a notification. |
+
+## Properties
+|Name | Type | Description|
+|:----|:-----|:-----------|
+| targetHostName | String | Represents the host name of the app to which the calling service wants to post the notification, for the given user. |
+| appNotificationId | String | This is the unique id set by the app server of a notification that is used to identify and target an individual notification. |
+| expirationDateTime | DateTimeOffset | This sets a UTC expiration time on a user notification - when time is up, the notification is removed from the Microsoft Graph notification feed store completely and is no longer part of notification history. Max value is 30 days. |
+| payload | Edm.ComplexType, JSON object | This is the data content of a raw or visual user notification that will be delivered to and consumed by the app client receiving this notification. |
```
Comment (yau-37) [inline api-reference/beta/resources/projectrome_notification.md]: Maybe we should add what happens when both a raw notification and a visual notification gets put in the payload at the same time

File: api-reference/beta/resources/projectrome_notification.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+| payload.visual.body | String | This is the body of a visual user notification. *Must have either title or body. |
+| displayTimeToLive | Int | This sets how long (in seconds) this notification content will stay in each platform’s notification viewer. For example, when the notification is delivered to a Windows device, the value of this property is passed on to ToastNotification.ExpirationTime, which determines how long the toast notification will stay in user’s Windows Action Center. |
+| priority | EnumType | This indicates the priority of a raw user notification. Visual notifications are sent with high priority by default. Valid values are High and Low. |
+| groupName | String | This is the name of the group that this notification belongs to. It is set by the developer for the purpose of grouping notifications together. |
+| targetPolicy | Edm.ComplexType, JSON object | Target policy object handles notification delivery policy at 2 different levels - endpoint types (Windows, iOS and Android) that should be targeted, and specific endpoints (identified by subscription ids) that should be targeted. |
```
Comment (yau-37) [inline api-reference/beta/resources/projectrome_notification.md]: We don't support sending to specific endpoints (subscription Ids) through MS Graph

File: api-reference/beta/api/projectrome_post_notification.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](../../../concepts/permissions_reference.md).
+
+|Permission type      | Permissions (from least to most privileged)              |
+|:--------------------|:---------------------------------------------------------|
+|Delegated (work or school account) | Notifications.ReadWrite.CreatedByApp    |
+|Delegated (personal Microsoft account) | Notifications.ReadWrite.CreatedByApp    |
+
+## HTTP request
+```http
+POST /me/notifications/}
```
Comment (yau-37) [inline api-reference/beta/api/projectrome_post_notification.md]: Remove end curly brace maybe? I'm not sure the convention here

File: api-reference/beta/api/projectrome_post_notification.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+## Example
+#### Request
+The following is an example of the request.
+
+```http
+POST https://graph.microsoft.com/beta/me/activities/
+Content-type: application/json
+
+{
+    "targetHostName": "undemo.activity.windows.com",
```
Comment (yau-37) [inline api-reference/beta/api/projectrome_post_notification.md]: Change undemo.activity.windows.com to client team's proposed host name for demo application

File: api-reference/beta/api/projectrome_post_notification.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    "payload": {
+        "visualContent": {
+            "title": "Hello Ignite!",
+            "body": "Notifications are Great!"
+        }
+    },
+    "targetPolicy": {
+        "platformTypes": [
+        "windows"
+        ]
```
Comment (yau-37) [inline api-reference/beta/api/projectrome_post_notification.md]: Add "ios" and/or "android" to add more understanding of how the feature may be used

File: api-reference/beta/api/projectrome_post_notification.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    },
+    "priority": "High",
+    "groupName": "TestGroup",
+    "displayTimeToLive": "23"
+}
+```
+
+#### Response
```
Comment (yau-37) [inline api-reference/beta/api/projectrome_post_notification.md]: Edit Response based on changes above along with removing personal info

File: api-reference/beta/resources/projectrome_notification.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+| targetPolicy.platformTypes | Edm.ComplexType, Collection (EnumType) | App developer can use this property if they want to filter the notification fanout to a specific platform(s). By default, all push endpoint types (iOS, Windows and Android) are enabled. |
+
+## Relationships
+None.
+
+## JSON representation
+Here is a JSON representation of the resource.
+```json
+{
```
Comment (yau-37) [inline api-reference/beta/resources/projectrome_notification.md]: The formatting of this JSON snippet when viewed is a little off

File: api-reference/beta/api/projectrome_post_notification.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+In the request body, supply a JSON representation of an [notification](../resources/projectrome_notification.md) object.
+
+## Response
+If successful, this method returns the `201 Created` response code if the notification was successfully created and stored. 
+## Example
+#### Request
+The following is an example of the request.
+
+```http
+POST https://graph.microsoft.com/beta/me/activities/
```
Comment (johndipu) [inline api-reference/beta/api/projectrome_post_notification.md]: me/notifications

File: api-reference/beta/resources/projectrome_notification.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+| expirationDateTime | DateTimeOffset | This sets a UTC expiration time on a user notification - when time is up, the notification is removed from the Microsoft Graph notification feed store completely and is no longer part of notification history. Max value is 30 days. |
+| payload | Edm.ComplexType, JSON object | This is the data content of a raw or visual user notification that will be delivered to and consumed by the app client receiving this notification. |
+| payload.rawContent | String | This is the notification content of a raw user notification that will be delivered to and consumed by the app client receiving this notification. *At least one of Payload.RawContent and Payload.VisualContent needs to be valid for a POST Notification request. |
+| payload.visual | Edm.ComplexType, JSON object | This is the visual content of a visual user notification, which will be directly consumed by notification platform on each mobile platform and rendered for the users. *At least one of Content and VisualContent needs to be valid for a POST Notification request. |
```
Comment (yau-37) [inline api-reference/beta/resources/projectrome_notification.md]: payload.visualContent.  Same with the two fields below relating to visual notifications

File: api-reference/beta/resources/projectrome_notification.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+When a user engages and acts on a visual notification, the app client can then use client-side Rome SDK to update the state of the corresponding notification feed in Microsoft Graph, such as, marking a notification as dismissed. The update will then be fanned-out to all other app client endpoints, so these clients can handle this change accordingly, such as, dismiss the same notification universally to prevent the user from seeing any redundant information. The same notification resource can also be accessed at a later time before it expires (even after being marked as dismissed), as notification history, by app clients through the Project Rome SDK. 
+
+## Methods
+|Method | Return Type | Description|
+|:------|:------------|:-----------|
+|[Create notification](../api/projectrome_post_notification.md) | [notification](projectrome_notification.md) |Creates and publish a notification. |
+
+## Properties
+|Name | Type | Description|
```
Comment (yau-37) [inline api-reference/beta/resources/projectrome_notification.md]: I think we should add which properties are nullable and which properties are required in the POST request

File: api-reference/beta/api/projectrome_post_notification.md [hunk 11]
```diff
@@ -0,0 +1,78 @@
+# Create and publish a user-centric notification to Microsoft Graph
+> **Important:** APIs under the /beta version in Microsoft Graph are in preview and are subject to change. Use of these APIs in production applications is not supported.
+Create and publish a user-centric notification to Microsoft Graph. The notification gets stored inside notification feed store in the Graph, and then gets fanned-out to all app clients on different device endpoints the user is logged in on.  
+## Permissions
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](../../../concepts/permissions_reference.md).
+
+|Permission type      | Permissions (from least to most privileged)              |
```
Comment (johndipu) [inline api-reference/beta/api/projectrome_post_notification.md]: Not sure if "(from least to most privileged)" adds any value here since the only 2 permissions we have now have the same privilege

File: api-reference/beta/api/projectrome_post_notification.md [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+|Delegated (personal Microsoft account) | Notifications.ReadWrite.CreatedByApp    |
+
+## HTTP request
+```http
+POST /me/notifications/}
+```
+## Request headers
+|Name | Type | Description|
+|:----|:-----|:-----------|
+|Authorization | string |The authorization header is used to pass the credentials of the calling party. OAuth: Authorization: bearer <access_token> |
```
Comment (johndipu) [inline api-reference/beta/api/projectrome_post_notification.md]: We should probably call out that this is a required header (although it should be fairly obvious)

Comment (johndipu) [inline api-reference/beta/api/projectrome_post_notification.md]: The "OAuth:" in the description feels incomplete...

File: api-reference/beta/api/projectrome_post_notification.md [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+## Request body
+In the request body, supply a JSON representation of an [notification](../resources/projectrome_notification.md) object.
```
Comment (johndipu) [inline api-reference/beta/api/projectrome_post_notification.md]: nit: typo - "a" notification

File: api-reference/beta/api/projectrome_post_notification.md [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+    "displayTimeToLive": "23"
+}
+```
+
+#### Response
+The following is an example of the response.
+
+```http
+HTTP/1.1 201
+Content-Type: application/json
```
[the remark this answers is not in the document]
Comment (yau-37) [reply api-reference/beta/api/projectrome_post_notification.md]: Location is indeed returned in the header

File: api-reference/beta/api/projectrome_post_notification.md [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+
+{
+    "@odata.context": "https://graph.microsoft.com/test872018/$metadata#users('rayxu1988%40gmail.com')/notifications/$entity",
+    "appNotificationId": "testDirectToastNotification",
+    "displayTimeToLive": 22,
```
[the remark this answers is not in the document]
Comment (yau-37) [reply api-reference/beta/api/projectrome_post_notification.md]: That's to be expected due to time processing in the back end of AFS

File: api-reference/beta/api/projectrome_post_notification.md [hunk 16]
```diff
@@ -0,0 +1,78 @@
+# Create and publish a user-centric notification to Microsoft Graph
```
Comment (Lauragra) [inline api-reference/beta/api/projectrome_post_notification.md]: If we're likely to change the Project Rome branding in the Ignite time frame, we'll want to update the file names. Technically the file name here should follow this convention: notification_post_notifications.

File: api-reference/beta/resources/projectrome_notification.md [hunk 17]
```diff
@@ -0,0 +1,60 @@
+# notification resource type
+> **Important:** APIs under the /beta version in Microsoft Graph are in preview and are subject to change. Use of these APIs in production applications is not supported.
+
+Represents a user-targeting notification that is published by app server. The notification is stored in Microsoft Graph and may be fanned-out to different device endpoints owned by this user. A notification can be a visual notification payload that can be directly interpreted by different OS platforms including Windows, Android, and iOS. It can also be a data payload which is delivered to and handled by app clients, which then determine the corresponding user experience on each device – usually, a visual notification UI is generated locally, that’s corresponding to the content in the original data payload.
```
Comment (johndipu) [inline api-reference/beta/resources/projectrome_notification.md]: Why do we say "**may** be fanned-out" here?

File: api-reference/beta/api/projectrome_post_notification.md [hunk 18]
```diff
@@ -0,0 +1,78 @@
...
+> **Important:** APIs under the /beta version in Microsoft Graph are in preview and are subject to change. Use of these APIs in production applications is not supported.
+Create and publish a user-centric notification to Microsoft Graph. The notification is stored in the Microsoft Graph notification feed store, and is sent to all app clients on all device endpoints that the user is signed in to.
```
Comment (Lauragra) [inline api-reference/beta/api/projectrome_post_notification.md]: What do we mean by "user-centric"? Later you also use the phrase "user-targeting". Can we simplify and just say "Create and send a notification to a user"? If so you'll want to update the title to "Create and send notification".

File: api-reference/beta/api/projectrome_post_notification.md (hunk 12 above)
Comment (Lauragra) [reply api-reference/beta/api/projectrome_post_notification.md]: The standard text for the description is: Bearer {token}. Required. Note that we use curly brackets, not angle brackets, as the angle brackets are dropped in the HTML conversion.

File: api-reference/beta/api/projectrome_post_notification.md [hunk 19]
```diff
@@ -0,0 +1,78 @@
...
+POST https://graph.microsoft.com/beta/me/activities/
+Content-type: application/json
+
+{
+    "targetHostName": "undemo.activity.windows.com",
+    "appNotificationId": "testDirectToastNotification",
+    "expirationDateTime": "2018-08-29T23:51:33.000Z",
+    "payload": {
+        "visualContent": {
+            "title": "Hello Ignite!",
```
Comment (Lauragra) [inline api-reference/beta/api/projectrome_post_notification.md]: Do you want an Ignite-specific example? Maybe change to something more generic.

File: api-reference/beta/resources/projectrome_notification.md (hunk 1 above)
Comment (yau-37) [reply api-reference/beta/resources/projectrome_notification.md]: Edit: We do not support this scenario of sending both raw and visual at the same time.  Please remove sample request w/ both values

File: api-reference/beta/resources/project_rome_overview.md [hunk 20]
```diff
@@ -42,3 +42,12 @@ You can use the following Microsoft Graph APIs to communicate with other Windows
 - [List the user's devices](../api/user_list_devices.md)
 - [Send a command to a device](../api/send_device_command.md)
 - [Get command status](../api/get_device_command_status.md)
+
+## Notifications
+
+You can use the notifications APIs in Microsoft Graph to deliver and manage notifications across multiple endpoints owned by the signed in user, in a human-centric, rather than a device-centric way. 
+
```
Comment (Lauragra) [inline api-reference/beta/resources/project_rome_overview.md]: I'm not sure that the phrase "in a human-centric, rather than a device-centric way" is completely clear. You might need to define "human-centric" and "device-centric".  Also, do you mean to say that the devices are "owned" by the user, or just that the user is signed in on the device? Would it be accurate to rephrase as "... across multiple endpoints that the user is signed in on, rather than across users' devices"?

File: api-reference/beta/resources/project_rome_overview.md [hunk 21]
```diff
@@ -42,3 +42,12 @@ You can use the following Microsoft Graph APIs to communicate with other Windows
...
+You can publish a raw data notification or a direct visual notification. When a raw data notification is delivered to a device endpoint, you can then use the client SDK (Graph notifications SDK for Windows, Project Rome SDK for iOS and Android) to receive and manage notifications. When a direct visual notification is delivered to a device endpoint, it shows the platform-specific native notification to the user. 
+
```
Comment (Lauragra) [inline api-reference/beta/resources/project_rome_overview.md]: Please clarify the SDK references here. Do you mean to reference a Microsoft Graph SDK listed on this page: https://developer.microsoft.com/en-us/graph/code-samples-and-sdks ?

File: api-reference/beta/resources/projectrome-graph-notifications-api-overview.md [hunk 22]
```diff
@@ -0,0 +1,7 @@
+# Use the MS Graph notifications REST API
```
Comment (Lauragra) [inline api-reference/beta/resources/projectrome-graph-notifications-api-overview.md]: Let's change the file name to notifications-api-overview, and the title to "Use the notifications REST API in Microsoft Graph".

File: api-reference/beta/resources/projectrome-graph-notifications-api-overview.md [hunk 23]
```diff
@@ -0,0 +1,7 @@
...
+This type of user-centric notification is represented by the [notification](https://developer.microsoft.com/en-us/graph/docs/api-reference/beta/resources/projectrome_notification) resource and are stored in Microsoft Graph, which can then be accessed and managed by the publishing app through the client-side Rome SDK APIs. 
+
+## Next steps
+- See the [notification resource](https://developer.microsoft.com/en-us/graph/docs/api-reference/beta/resources/projectrome_notification) and create notifications to engage with your users. 
+- (Coming soon) Try the API in the [Graph Explorer](https://developer.microsoft.com/en-us/graph/graph-explorer).
```
Comment (Lauragra) [inline api-reference/beta/resources/projectrome-graph-notifications-api-overview.md]: What is the ETA for adding notifications to Graph Explorer? If before we'll publish the docs, please remove "Coming soon". If after, let's remove this bullet for now and add it when the sample queries are available.

File: api-reference/beta/resources/projectrome-graph-notifications-api-overview.md [hunk 24]
```diff
@@ -0,0 +1,8 @@
+# Use the MS Graph notifications REST API
+
+You can use the notifications API in Microsoft Graph to send push notifications to a user. Simply target a user account to publish a notification to, and the platform will help you to fan-out and deliver this notification to all device endpoints. Notifications API requests are performed on behalf of a user via [delegated permissions](../../../concepts/permissions_reference.md#delegated-permissions-application-permissions-and-effective-permissions) and the [notification permission]( ../../../concepts/permissions_reference.md), which can be used with either Microsoft accounts or work or school accounts.
+This type of user-centric notification is represented by the [notification](..resources/projectrome_notification.md) resource and is stored in Microsoft Graph. It can then be accessed and managed by the publishing app through the client-side Rome SDK APIs. 
+
```
Comment (Lauragra) [inline api-reference/beta/resources/projectrome-graph-notifications-api-overview.md]: Please clarify what SDKs you're referencing. Would be good to provide a direct link.

File: api-reference/beta/resources/projectrome-graph-notifications-api-overview.md [hunk 25]
```diff
@@ -0,0 +1,8 @@
...
+## Next steps
+- See the [notification resource](../resources/projectrome_notification.md) and create notifications to engage with your users. 
+- (Coming soon) Try the API in the [Graph Explorer](https://developer.microsoft.com/en-us/graph/graph-explorer).
```
Comment (Lauragra) [inline api-reference/beta/resources/projectrome-graph-notifications-api-overview.md]: What is the ETA for adding notifications to Graph Explorer? If before we'll publish the docs, please remove "Coming soon". If after, let's remove this bullet for now and add it when the sample queries are available.

File: api-reference/beta/resources/projectrome-graph-notifications-api-overview.md [hunk 26]
```diff
@@ -0,0 +1,8 @@
+# Use the MS Graph notifications REST API
+
+You can use the notifications API in Microsoft Graph to send push notifications to a user. Simply target a user account to publish a notification to, and the platform will help you to fan-out and deliver this notification to all device endpoints. Notifications API requests are performed on behalf of a user via [delegated permissions](../../../concepts/permissions_reference.md#delegated-permissions-application-permissions-and-effective-permissions) and the [notification permission]( ../../../concepts/permissions_reference.md), which can be used with either Microsoft accounts or work or school accounts.
```
Comment (Lauragra) [inline api-reference/beta/resources/projectrome-graph-notifications-api-overview.md]: The phrase "the platform will help you to fan-out and deliver this notification" is unclear. Would it be accurate to revise as "the platform delivers the notification"?

File: api-reference/beta/resources/projectrome_notification.md (hunk 17 above)
Comment (Lauragra) [inline api-reference/beta/resources/projectrome_notification.md]: Please clarify what you mean by "user-targeting". Can you simply call it a "user notification"?

File: api-reference/beta/resources/projectrome_notification.md [hunk 27]
```diff
@@ -0,0 +1,60 @@
+# notification resource type
+> **Important:** APIs under the /beta version in Microsoft Graph are in preview and are subject to change. Use of these APIs in production applications is not supported.
+
+Represents a user-targeting notification that is published by an app server. The notification is stored in Microsoft Graph and may be fanned-out to different device endpoints owned by this user. A notification can be a visual notification payload that can be directly interpreted by different OS platforms including Windows, Android, and iOS. It can also be a data payload which is delivered to and handled by app clients, which then determine the corresponding user experience on each device – usually, a visual notification UI is generated locally, that’s corresponding to the content in the original data payload.
```
Comment (Lauragra) [inline api-reference/beta/resources/projectrome_notification.md]: Please clarify what you mean by "user-targeting notification". Can you simply call it a "user notification"?

 @Lauragra	Reply…
 When a user engages and acts on a visual notification, the app client can then use client-side Rome SDK to update the state of the corresponding notification feed in Microsoft Graph, such as, marking a notification as dismissed. The update will then be fanned-out to all other app client endpoints, so these clients can handle this change accordingly, such as, dismiss the same notification universally to prevent the user from seeing any redundant information. The same notification resource can also be accessed at a later time before it expires (even after being marked as dismissed), as notification history, by app clients through the Project Rome SDK. 
 ## Methods
|Method | Return Type | Description|
|:------|:------------|:-----------|
|[Create notification](../api/projectrome_post_notification.md) | [notification](projectrome_notification.md) |Creates and publish a notification. |
 ## Properties
|Name | Type | Description|
  @yau-37
yau-37 3 days ago
I think we should add which properties are nullable and which properties are required in the POST request

 @Lauragra	Reply…
 |:----|:-----|:-----------|
| targetHostName | String | Represents the host name of the app to which the calling service wants to post the notification, for the given user. |
| appNotificationId | String | This is the unique id set by the app server of a notification that is used to identify and target an individual notification. |
| expirationDateTime | DateTimeOffset | This sets a UTC expiration time on a user notification - when time is up, the notification is removed from the Microsoft Graph notification feed store completely and is no longer part of notification history. Max value is 30 days. |
| payload | Edm.ComplexType, JSON object | This is the data content of a raw or visual user notification that will be delivered to and consumed by the app client receiving this notification. |
  @yau-37
yau-37 3 days ago
Maybe we should add what happens when both a raw notification and a visual notification gets put in the payload at the same time

  @yau-37
yau-37 5 hours ago
Edit: We do not support this scenario of sending both raw and visual at the same time. Please remove sample request w/ both values

 @Lauragra	Reply…
 | payload.rawContent | String | This is the notification content of a raw user notification that will be delivered to and consumed by the app client receiving this notification. *At least one of Payload.RawContent and Payload.VisualContent needs to be valid for a POST Notification request. |
| payload.visual | Edm.ComplexType, JSON object | This is the visual content of a visual user notification, which will be directly consumed by notification platform on each mobile platform and rendered for the users. *At least one of Content and VisualContent needs to be valid for a POST Notification request. |
  @yau-37
yau-37 3 days ago
payload.visualContent. Same with the two fields below relating to visual notifications

 @Lauragra	Reply…
 | payload.visual.title | String | This is the title of a visual user notification. *Must have either title or body. |
| payload.visual.body | String | This is the body of a visual user notification. *Must have either title or body. |
| displayTimeToLive | Int | This sets how long (in seconds) this notification content will stay in each platform’s notification viewer. For example, when the notification is delivered to a Windows device, the value of this property is passed on to ToastNotification.ExpirationTime, which determines how long the toast notification will stay in user’s Windows Action Center. |
| priority | EnumType | This indicates the priority of a raw user notification. Visual notifications are sent with high priority by default. Valid values are High and Low. |
| groupName | String | This is the name of the group that this notification belongs to. It is set by the developer for the purpose of grouping notifications together. |
| targetPolicy | Edm.ComplexType, JSON object | Target policy object handles notification delivery policy at 2 different levels - endpoint types (Windows, iOS and Android) that should be targeted, and specific endpoints (identified by subscription ids) that should be targeted. |
  @yau-37
yau-37 3 days ago
We don't support sending to specific endpoints (subscription Ids) through MS Graph

 @Lauragra	Reply…
 | targetPolicy.platformTypes | Edm.ComplexType, Collection (EnumType) | App developer can use this property if they want to filter the notification fanout to a specific platform(s). By default, all push endpoint types (iOS, Windows and Android) are enabled. |
 ## Relationships
None.
 ## JSON representation
Here is a JSON representation of the resource.
```json
{	
  @yau-37
yau-37 3 days ago
The formatting of this JSON snippet when viewed is a little off

 @Lauragra	Reply…
 	"targetHostName": "String",
    "appNotificationId": "String",
	"expirationDateTime": "DateTimeOffset",
	"payload":  [
	{
		"rawContent": "String",
		"visual": [
			{
				"title": "String",
				"body": "String"
			}
		],
	}
],
    "displayTimeToLive": "Int",
    "priority": "Enum",
	"groupName": "String",
	"targetPolicy":  [
	{
		"platformTypes": [{ "@odata.type": "Enum" }]
	}
],
}
```

File: api-reference/beta/resources/projectrome_notification.md [hunk 28]
```diff
@@ -0,0 +1,60 @@
+# notification resource type
+> **Important:** APIs under the /beta version in Microsoft Graph are in preview and are subject to change. Use of these APIs in production applications is not supported.
+
+Represents a user-targeting notification that is published by an app server. The notification is stored in Microsoft Graph and is fanned out to different device endpoints owned by this user. A notification can be a visual notification payload that can be directly interpreted by different operating systems, including Windows, Android, and iOS. It can also be a data payload that's delivered to and handled by app clients, which then determine the corresponding user experience on each device – usually, a visual notification UI that corresponds to the content in the original data payload is generated locally.
```
Comment (Lauragra) [inline api-reference/beta/resources/projectrome_notification.md]: Please clarify "user-targeting" and whether we can just call it a "user notification".

Comment (Lauragra) [inline api-reference/beta/resources/projectrome_notification.md]: Instead of "fanned out", can we just say "distributed"?

File: api-reference/beta/resources/projectrome_notification.md [hunk 29]
```diff
@@ -0,0 +1,60 @@
+# notification resource type
+> **Important:** APIs under the /beta version in Microsoft Graph are in preview and are subject to change. Use of these APIs in production applications is not supported.
+
+Represents a user-targeting notification that is published by an app server. The notification is stored in Microsoft Graph and is fanned out to different device endpoints owned by the user. A notification can be a visual notification payload that can be interpreted by the operating system, including Windows, Android, and iOS platforms. It can also be a data payload that's delivered to and handled by app clients, which then determine the corresponding user experience on each device – usually, a visual notification UI that corresponds to the content in the original data payload that's generated locally.
```
Comment (Lauragra) [inline api-reference/beta/resources/projectrome_notification.md]: Please clarify "user-targeting" and whether we can just call it a "user notification".

Comment (Lauragra) [inline api-reference/beta/resources/projectrome_notification.md]: Instead of "fanned out", can we just say distributed?

File: api-reference/beta/resources/projectrome_notification.md [hunk 30]
```diff
@@ -0,0 +1,60 @@
...
+When a user acts on a visual notification, the app client can then use client-side Rome SDK to update the state of the corresponding notification feed in Microsoft Graph - for example, by marking a notification as dismissed. The update will then be fanned out to all other app client endpoints, and the clients handle the change accordingly, for example by dismissing the notification to prevent the user from seeing redundant information. App clients can access the same notification resource at a later time before it expires (even after it is marked as dismissed), as notification history, via the Project Rome SDK.
```
Comment (Lauragra) [inline api-reference/beta/resources/projectrome_notification.md]: Please clarify the SDK that you're referring to and refer to it consistently throughout.

File: api-reference/beta/resources/projectrome_notification.md [hunk 31]
```diff
@@ -0,0 +1,60 @@
...
+| priority | EnumType | Indicates the priority of a raw user notification. Visual notifications are sent with high priority by default. Valid values are High and Low. |
+| groupName | String | The name of the group that this notification belongs to. It is set by the developer for the purpose of grouping notifications together. |
+| targetPolicy | Edm.ComplexType, JSON object | Target policy object handles notification delivery policy at two different levels - endpoint types (Windows, iOS and Android) that should be targeted, and specific endpoints (identified by subscription ids) that should be targeted. |
+| targetPolicy.platformTypes | Edm.ComplexType, Collection (EnumType) | Use to filter the notification fanout to a specific platform or platforms. By default, all push endpoint types (iOS, Windows, and Android) are enabled. |
```
Comment (Lauragra) [inline api-reference/beta/resources/projectrome_notification.md]: Instead of "fanout", can you say "distribution"?

File: api-reference/v1.0/resources/project_rome_overview.md [hunk 32]
```diff
@@ -2,10 +2,10 @@
 
 [Project Rome](https://developer.microsoft.com/en-us/windows/project-rome) is a Microsoft initiative to build a cross-device experiences platform. Project Rome enables an app on a local client or service to interact with apps and services on a remote host when the user signs in with the same Microsoft account that they use to sign in on the client device. This allows you to program cross-device and cross-platform experiences that are centered around user tasks rather than devices.
 
-A key component is exposed via Microsoft Graph to enable these experiences: activities.
-
 ## Activities
 
+A key component is exposed via Microsoft Graph to enable these experiences: activities.
+
```
Comment (Lauragra) [inline api-reference/v1.0/resources/project_rome_overview.md]: This text should stay above the H2. It looks like you edited the v1.0 version of the file. Please revert this edit and update the beta version of the same file. In that file, you'll change "Two key components" to "Three key components" and add notifications after the colon. Then add the H2 for Notifications.

File: api-reference/v1.0/resources/project_rome_overview.md [hunk 33]
```diff
@@ -23,4 +23,8 @@ You can use the following Microsoft Graph APIs to create and retrieve user activ
 - [Get recent activities](../api/projectrome_get_recent_activities.md)
 - [Delete an activity](../api/projectrome_delete_activity.md)
 - [Create or replace a history item](../api/projectrome_put_historyitem.md)
-- [Delete a history item](../api/projectrome_delete_historyitem.md)
\ No newline at end of file
+- [Delete a history item](../api/projectrome_delete_historyitem.md)
+
+## Notifications 
+
+You can use the notifications API in Microsoft Graph to enable your app to send human-centric notifications, or to improve your exisitng notification strategy to target users. The notifications API handles the mapping between users and endpoints, syncs notification state across users' devices, and more.
```
Comment (Lauragra) [inline api-reference/v1.0/resources/project_rome_overview.md]: Please clarify "human-centric".

File: api-reference/beta/resources/projectrome_notification.md [hunk 34]
```diff
@@ -0,0 +1,60 @@
...
+Here is a JSON representation of the resource.
+```json
+{	
+	"targetHostName": "String",
+    "appNotificationId": "String",
+	"expirationDateTime": "DateTimeOffset",
+	"payload":  [
+	{
+		"rawContent": "String",
+		"visual": [
```
Comment (yau-37) [inline api-reference/beta/resources/projectrome_notification.md]: This should be visualContent

File: api-reference/beta/api/projectrome_post_notification.md (hunk 16 above)
Comment (leixu2046) [reply api-reference/beta/api/projectrome_post_notification.md]: There's no plan to change the branding by Ignite

File: api-reference/beta/resources/project_rome_overview.md (hunk 20 above)
Comment (leixu2046) [reply api-reference/beta/resources/project_rome_overview.md]: Human-centric and device-centric are terms we've been using across Rome docs. I've rephrased it to make it clear tho

File: api-reference/beta/resources/project_rome_overview.md (hunk 21 above)
Comment (leixu2046) [reply api-reference/beta/resources/project_rome_overview.md]: No its the project Rome SDKs here: https://github.com/Microsoft/project-rome and I've updated the doc.

File: api-reference/beta/resources/projectrome-graph-notifications-api-overview.md (hunk 25 above)
Comment (leixu2046) [reply api-reference/beta/resources/projectrome-graph-notifications-api-overview.md]: The publication of the docs is required for notifications to show up inside the Graph Explorer. But we will make it go live immediately after the docs are published (targeting 17th). So I removed "coming soon", but I kept this link.

File: api-reference/beta/resources/project_rome_overview.md [hunk 35]
```diff
@@ -42,3 +42,12 @@ You can use the following Microsoft Graph APIs to communicate with other Windows
 - [List the user's devices](../api/user_list_devices.md)
 - [Send a command to a device](../api/send_device_command.md)
 - [Get command status](../api/get_device_command_status.md)
+
+## Notifications
+
+You can use the notifications APIs in Microsoft Graph to deliver notifications across multiple endpoints that the same user is signed in on. You can target a user directly when posting notifications instead of worrying about device addresses/channels. This way, you can focus on designing the right notification scenarios in a human-centric, rather than a device-centric way. 
+
+You can publish a raw data notification or a direct visual notification. When a raw data notification is delivered to a device endpoint, you can then use the [client SDK](https://github.com/Microsoft/project-rome) (Graph notifications SDK for Windows, Project Rome SDK for iOS and Android) to receive and manage notifications. When a direct visual notification is delivered to a device endpoint, it shows the platform-specific native notification to the user. 
+
```
Comment (Lauragra) [inline api-reference/beta/resources/project_rome_overview.md]: Please verify the link to the SDKs, as some of the folders only appear to include samples. Is there a "Microsoft Graph notifications SDK for Windows"?

Comment (leixu2046) [reply api-reference/beta/resources/project_rome_overview.md]: Hi Laura there is one it's not uploaded yet. We will upload and push a change here. We will likely add a text file in this location to point to Windows NuGet package. So worst case scenario just having this single link in this page is still safe.

File: api-reference/v1.0/resources/project_rome_overview.md (hunk 32 above)
Comment (leixu2046) [reply api-reference/v1.0/resources/project_rome_overview.md]: Done.

File: api-reference/beta/resources/projectrome_notification.md [hunk 36]
```diff
@@ -0,0 +1,80 @@
...
+When a user acts on a visual notification, the app client can then use client-side Project Rome SDK to update the state of the corresponding notification feed in Microsoft Graph - for example, by marking a notification as dismissed. The update will then be fanned out to all other app client endpoints, and the clients handle the change accordingly, for example by dismissing the notification to prevent the user from seeing redundant information. App clients can access the same notification resource at a later time before it expires (even after it is marked as dismissed), as notification history, via the [Project Rome SDK](https://github.com/Microsoft/project-rome). 
+
```
Comment (Lauragra) [inline api-reference/beta/resources/projectrome_notification.md]: Instead of "fanned out", change to "distributed"?
